nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/flow/StandardFlowManager.java

```diff
@@ -519,6 +524,74 @@ public ReportingTaskNode createReportingTask(final String type, final String id,
         return taskNode;
     }
 
+    @Override
+    public FlowAnalysisRuleNode createFlowAnalysisRule(
+            String type,
+            String id,
+            BundleCoordinate bundleCoordinate,
+            Set<URL> additionalUrls,
+            boolean firstTimeAdded,
+            boolean register,
+            String classloaderIsolationKey
+    ) {
+        if (type == null || id == null || bundleCoordinate == null) {
+            throw new NullPointerException();
```

Comment:
An empty `NullPointerException` usually  connotes an error that should be corrected with better code handling. Recommend using `IllegalArgumentException` with a message, or `Objects.requireNonNull()` with a message.